Dimmer class configuration methods and some additional control methods

PaulWieland opened the pull request.

PaulWieland committed:
Added dimmer control methods, configuration settings and some new events

PaulWieland committed:
Added emitter string to debug log

PaulWieland committed:
Converting comment to english

PaulWieland commented on src/devices/DimmableLightingDevice/DimmableLightingDevice.ts:
```diff
@@ -125,6 +69,7 @@ export default class DimmableLightingDevice extends InsteonDevice {
 					}
 					break;
 				case 0x18: this.emitRemote(['dim','stoppedChanging'], data); break;
+				case 0x21: this.emitRemote(['switch','instant'], data); break;
```
Not sure if this should be `'switch','instant'` or `'switch','on'` or `'switch','on','instant'`. I'm not sure if you would ever want to subscribe to just the instant event.

apearson replied:
`'switch','on','instant'`

With eventemitter if you sub to `'switch', 'on'` it should subscribe you to everything under it as well

PaulWieland replied:
Events renamed

PaulWieland commented on src/devices/DimmableLightingDevice/DimmableLightingDevice.ts:
```diff
@@ -136,24 +81,224 @@ export default class DimmableLightingDevice extends InsteonDevice {
 
 		// Parsing status out
 		return {
+			// state: statusPacket.cmd === 0x00 ? "off" : "on",
```
Was contemplating adding the state `on|off` in addition to the level.

apearson replied:
I think just the level would be fine.  The user can do the calculation on their own

PaulWieland commented on src/devices/DimmableLightingDevice/DimmableLightingDevice.ts:
```diff
@@ -216,6 +361,38 @@ export default class DimmableLightingDevice extends InsteonDevice {
 		return this.sendInsteonCommand(0x18,0x00);
 	}
 	
+	/* Not yet tested. Based soley on documentation */
```
Will test all three of these as soon as possible - I believe the last batch of dimmer switches that I bought support the light on at rate commands.

apearson replied:
I can test these on my dimmers if needed

apearson commented on src/devices/InsteonDevice.ts:
```diff
@@ -446,7 +446,7 @@ export default class InsteonDevice extends EventEmitter2 {
 			            : 'U';
 
 			if(this.options.debug)
-				console.log(`[←][${this.addressString}][${pType}][${data.Flags.Subtype}]: ${toHex(data.cmd1)} ${toHex(data.cmd2)} ${(data.extendedData || []).map(toHex)}`);
```
Let's not change the debug statement

apearson reviewed (changes requested):
I've left some comments with changes I'd like.  Let me know if any changes need more discussion

PaulWieland committed:
level is enough

PaulWieland committed:
better event names

PaulWieland committed:
Switched load sense bit.

I suspect the docs are wrong on which bit is for storing the load sense capability. Further testing is required to confirm.

Also the `setLEDEnabled` command was doing the opposite of what it said, therefore its been renamed to `setLEDDisabled`

PaulWieland committed:
Comments

PaulWieland committed:
Added events & configuration get/set

PaulWieland committed:
Used `config` instead of `configuration` to keep consistency with other methods

PaulWieland committed:
Used `config` instead of `configuration` to keep consistency with other methods

PaulWieland committed:
Added configuration methods for the motion sensor

PaulWieland committed:
Light Sensitivity makes more sense

PaulWieland committed:
Added clamp utility per issue #23

https://github.com/apearson/insteon-plm/issues/23

PaulWieland committed:
Renamed modes for consistency sake

PaulWieland committed:
Corrected bit order for motion sensor config

Tested & works
